daemon/src/rollover/protocol.rs

```diff
@@ -141,3 +190,402 @@ pub(crate) async fn emit_failed(order_id: OrderId, e: anyhow::Error, executor: &
         tracing::error!(%order_id, "Failed to execute rollover failed: {e:#}")
     }
 }
+
+pub(crate) fn build_punish_params(
+    our_role: Role,
+    sk: SecretKey,
+    counterparty_pk: PublicKey,
+    other_msg0: RolloverMsg0,
+    rev_pk: bdk::bitcoin::PublicKey,
+    publish_pk: bdk::bitcoin::PublicKey,
+) -> (
+    (PublicKey, PunishParams),
```

Comment:
It might be worth introducing different types at some point, 
e.g. 

```suggestion
    MakerPunishParams,
    TakerPunishParams, ...
```

Reply:
Happy to see this in a follow-up

Reply:
In an effort to have more assurance that we did not accidentally flip something I added https://github.com/itchysats/itchysats/pull/2248/commits/b644793afccb61552e2061f9121669af9381d313

Note: The `double_rollover_test` added recently is a good safety net - during the refactoring I did run into a flip and the test failed on the second run with the `Counterparty sent invalid revocation sk` ;)